HTTP Server.core/IFunctionality.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace HTTPServer.core
+{
+    public interface IFunctionality
+    {
+        byte[] Execute(Request request);
+        bool ShouldRun(Request request, IPathContents pathContents);
```

Comment:
I think there's a simpler interface to be had than `Execute` and `ShouldRun` (specifically regarding the `ShouldRun`). Why do you need the `ShouldRun` method?